micropy/project/modules/packages.py

```diff
@@ -138,28 +118,29 @@ def add_package(self, package, dev=False, **kwargs):
             dict: Dictionary of packages
 
         """
-        pkg = package
-        if isinstance(package, str):
-            pkg = next(requirements.parse(package))
+
+        source = create_dependency_source(package)
+        pkg = source.package
         self.log.info(f"Adding $[{pkg.name}] to requirements...")
-        specs = "".join(next(iter(pkg.specs))) if pkg.specs else "*"
         if self.packages.get(pkg.name, None):
-            self.log.error(f"$[{pkg.name}] is already installed!")
+            self.log.error(f"$[{pkg}] is already installed!")
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Instance of 'PackagesModule' has no 'log' member](https://app.codacy.com/app/BradenM/micropy-cli/pullRequest?prid=4593453)